### ISSUE TITLE
GPU usage

### ISSUE THREAD

**Issue by watanka** [Media-Smart/vedastr#18]
Anyone know how to activate GPU usage?

**Comment by watanka** [comment]
I set config in tps_resnet_bilstm_attn.py
1. 'gpu_id'
2. cudnn_deterministic = True
3. cudnn_benchmark = True

**Comment by ChaseMonsterAway** [comment]
> Anyone know how to activate GPU usage?

what do you mean 'gpu usage'?

**Comment by watanka** [comment]
thank you for such quick response.
It seems my gpu is not running while running the model, and the model is running real slow.

I was wondering if there was a config change that I should have to make in order to use gpu.

**Comment by ChaseMonsterAway** [comment]
Do you set the ``gpu_id``  correctly?

**Comment by ChaseMonsterAway** [comment]
Could you show me the log?

**Comment by watanka** [comment]
yup. I set 'gpu_id'= '3' (have 8 gpu)

2020-08-04 05:06:52,645 - INFO - Use CPU <---
2020-08-04 05:06:52,645 - INFO - Set cudnn deterministic True
2020-08-04 05:06:52,645 - INFO - Set cudnn benchmark True
2020-08-04 05:06:52,645 - INFO - Set seed 1111
2020-08-04 05:06:52,650 - INFO - Build model
2020-08-04 05:06:53,404 - INFO - GResNet init weights
2020-08-04 05:06:54,173 - INFO - AttHead init weights
2020-08-04 05:07:56,913 - INFO - current dataset length is 891924 in ../../../2_shared/data_lmdb_release/training/MJ/MJ_test
2020-08-04 05:08:59,225 - INFO - current dataset length is 802731 in ../../../2_shared/data_lmdb_release/training/MJ/MJ_valid
2020-08-04 05:18:11,649 - INFO - current dataset length is 7224586 in ../../../2_shared/data_lmdb_release/training/MJ/MJ_train
2020-08-04 05:23:31,557 - INFO - current dataset length is 5522807 in ../../../2_shared/data_lmdb_release/training/ST/
2020-08-04 05:23:58,478 - INFO - current dataset length is 6992 in ../../../2_shared/data_lmdb_release/validation/
2020-08-04 05:23:58,484 - INFO - Start train...
2020-08-04 06:09:22,621 - INFO - Train, Epoch 1, Iter 10, LR [1.0], Loss 2.9516, acc 0.0000, edit_distance 0.052872357561572436
2020-08-04 06:09:22,621 - INFO - 
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
tester                    | e                         | 0.0631	False
sheerest                  | e                         | 0.0647	False
imago                     | ee                        | 0.0065	False
mauro                     | e                         | 0.0677	False
classifieds               |                           | 0.0000	False
--------------------------------------------------------------------------------
2020-08-04 06:41:53,078 - INFO - Train, Epoch 1, Iter 20, LR [1.0], Loss 2.9336, acc 0.0010, edit_distance 0.08496536719592619
2020-08-04 06:41:53,079 - INFO - 
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
intended                  | t                         | 0.0814	False
poachers                  | ae                        | 0.0077	False
teach                     | aae                       | 0.0007	False
enthroned                 | ae                        | 0.0075	False
rosetta                   | ae                        | 0.0074	False
--------------------------------------------------------------------------------
2020-08-04 07:20:55,178 - INFO - Train, Epoch 1, Iter 30, LR [1.0], Loss 2.8503, acc 0.0007, edit_distance 0.09943515082550337
2020-08-04 07:20:55,179 - INFO - 
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
alberta                   | aee                       | 0.0013	False
stagger                   | aee                       | 0.0011	False
convectors                | aeee                      | 0.0001	False
signifies                 | aeee                      | 0.0001	False
recrudescence             | aee                       | 0.0012	False
--------------------------------------------------------------------------------
2020-08-04 07:57:53,674 - INFO - Train, Epoch 1, Iter 40, LR [1.0], Loss 2.8530, acc 0.0006, edit_distance 0.10911482519967489
2020-08-04 07:57:53,674 - INFO - 
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
capacities                | aaan                      | 0.0001	False
tanisha                   | aaae                      | 0.0001	False
janglers                  | aa                        | 0.0115	False
swastikas                 | aaoe                      | 0.0001	False
splatting                 | aaae                      | 0.0001	False
--------------------------------------------------------------------------------

**Comment by ChaseMonsterAway** [comment]
Do you install pytorch, cuda, cudnn correctly?  what is the results of ``torch.cuda.is_available()`` in you enviroment?

**Comment by watanka** [comment]
torch.cuda.is_available turns out to be False
updated torch version and now it's working.
Thank you for your help :)

**Comment by ChaseMonsterAway** [comment]
If you find our work useful, please star this repo. Thanks.
